eZ/Publish/Core/IO/Exception/InvalidBinaryFileIdException.php

```diff
@@ -8,12 +8,22 @@
  */
 namespace eZ\Publish\Core\IO\Exception;
 
+use Exception;
 use eZ\Publish\Core\Base\Exceptions\InvalidArgumentValue;
 
 class InvalidBinaryFileIdException extends InvalidArgumentValue
```

Comment:
Maybe I'd actually create a subclass for it:

```php
class InvalidBinaryPrefixException extends InvalidBinaryFileIdException
{
  public function __construct($id, $prefixSetting)
  {
    parent::__construct($id);
    
    $this->setMessageTemplate("Argument 'BinaryFile::id' is invalid: '%id%' is wrong value, it does not contain prefix '%prefix%'. Is 'var_dir' config correct?";
    $this->setParameters(['id' => $id, 'prefix' => $prefix]);

    Exception::__construct()
  }
}
```

It would make the `it does not contain prefix...` part translatable as well.